src/Symfony/Component/Notifier/Tests/ChatterTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Symfony\Component\Notifier\Tests;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Notifier\Chatter;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Tests\Transport\DummyMessage;
+use Symfony\Component\Notifier\Transport\TransportInterface;
+
+class ChatterTest extends TestCase
+{
+    /** @var MockObject&TransportInterface */
+    private $transport;
+
+    /** @var MockObject&MessageBusInterface */
+    private $bus;
+
+    protected function setUp(): void
+    {
+        $this->transport = $this->createMock(TransportInterface::class);
+        $this->bus = $this->createMock(MessageBusInterface::class);
+    }
+
+    public function testSendWithoutBus()
+    {
+        $chatter = new Chatter($this->transport);
+
+        $message = new DummyMessage();
+
+        $sentMessage = new SentMessage($message, 'any');
+
+        $this->transport
+            ->expects($this->once())
+            ->method('send')
+            ->with($message)
+            ->willReturn($sentMessage);
+
+        $this->assertSame($sentMessage, $chatter->send($message));
+        $this->assertSame($message, $sentMessage->getOriginalMessage());
+    }
+
+    public function testSendWithBus()
+    {
+        $chatter = new Chatter($this->transport, $this->bus);
+
+        $message = new DummyMessage();
+
+        $this->transport
+            ->expects($this->never())
+            ->method('send')
+            ->with($message);
+
+        $this->bus
+            ->expects($this->once())
+            ->method('dispatch')
+            ->with($message)
+            ->willReturn(new Envelope(new \stdClass()));
+        ;
```

Comment:
I know this works, but its reading a bit weird, I am always in favour of defining the expectations before injection/using them somewhere.

Can you please move this above ` $chatter = new Chatter($this->transport, $this->bus);` ?

Thanks

Same for the other tests